### PR TITLE
Feat: Added Folder/File structure for the video player section

### PR DESCRIPTION
**Pull request by yash2324**
Screenshot 2024-07-14 at 1 54 21 AM
Screenshot 2024-07-14 at 1 54 18 AM

### CODE REVIEW

File: desk-ui/src/app/components/FolderStructure.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    useState<expandedSectionData[]>();
+  const searchParams = useSearchParams();
+  const course_id = searchParams.get("course_id");
+  useEffect(() => {
+    const getCourse = async () => {
+      if (!course_id) return;
+
+      try {
+        const courseInfo = await fetch(
+          `http://127.0.0.1:8000/course/:id?id=${course_id}`,
```
Comment (ishar19) [inline desk-ui/src/app/components/FolderStructure.tsx]: Have the url as an environment variable

Comment (ishar19) [reply desk-ui/src/app/components/FolderStructure.tsx]: Make a api folder to make all the calls, keep api calls seperate

File: desk-ui/src/app/components/FolderStructure.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+          { method: "GET" }
+        );
+        const courseData = await courseInfo.json();
+        console.log(courseData);
```
Comment (ishar19) [inline desk-ui/src/app/components/FolderStructure.tsx]: Remove logs

**Review by ishar19 — CHANGES_REQUESTED** [review]

**Review by ishar19 — COMMENTED** [review]
Does it also works when we have folder inside a folder?

**Comment by yash2324** [comment]
> Does it also works when we have folder inside a folder?

No it is not working for that just checked it displays that as a lecture 
Rest of the issues have been fixed

File: desk-ui/src/app/api/Folders/[id]/route.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    const courseInfo = await fetch(`${BASE_URL}/course/:id?id=${id}`, {
+      method: "GET",
+    });
+
+    if (!courseInfo.ok) {
+      throw new Error(`HTTP error! status: ${courseInfo.status}`);
+    }
+
+    const courseData = await courseInfo.json();
+    console.log(courseData);
```
Comment (ishar19) [inline desk-ui/src/app/api/Folders/[id]/route.ts]: remove success logs, only keep error logs

File: server/app/api/controller/file_controller.py [hunk 4]
```diff
@@ -1,25 +1,71 @@
...
+import json
+import uuid
+from datetime import time
+from pathlib import Path
+from typing import List, Dict, Any
+
+from ...schemas.lesson import Lesson
+
+
+def determine_file_type(file_name):
```
Comment (ishar19) [inline server/app/api/controller/file_controller.py:13]: dont encapsulate file type, i need it for proper rendering

**Review by ishar19 — CHANGES_REQUESTED** [review]

**Comment by yash2324** [comment]
@ishar19 bhaiya please check now

**Comment by ishar19** [comment]
Badiya, merge krra hu mai

**Pull request merged**